### PR TITLE
[system-probe] Fix flaky test

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Fixes `TestTCPDirectionWithPreexistingConnection` after failure seen in CI.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/tracer_linux_test.go [hunk 1]
```diff
@@ -1708,13 +1707,18 @@ func TestTCPDirectionWithPreexistingConnection(t *testing.T) {
...
-	require.Len(t, origConn, 1)
+	require.Eventually(t, func() bool {
+		conns := getConnections(t, tr)
+		origConn = searchConnections(conns, func(cs network.ConnectionStats) bool {
+			return fmt.Sprintf("%s:%d", cs.Source, cs.SPort) == server.address &&
+				fmt.Sprintf("%s:%d", cs.Dest, cs.DPort) == c.LocalAddr().String()
+		})
+
+		return len(origConn) == 1
+	}, 3*time.Second, 500*time.Millisecond, "timed out waiting for original conneection")
```
Comment (mbakht) [inline pkg/network/tracer/tracer_linux_test.go:1720]: typo: conneection

**Review by mbakht — APPROVED** [review]

File: pkg/network/tracer/tracer_linux_test.go [hunk 2]
```diff
@@ -1708,13 +1707,18 @@ func TestTCPDirectionWithPreexistingConnection(t *testing.T) {
...
 	// the original connection should still be incoming for the server
-	conns := getConnections(t, tr)
-	origConn := searchConnections(conns, func(cs network.ConnectionStats) bool {
-		return fmt.Sprintf("%s:%d", cs.Source, cs.SPort) == server.address &&
-			fmt.Sprintf("%s:%d", cs.Dest, cs.DPort) == c.LocalAddr().String()
-	})
-	require.Len(t, origConn, 1)
+	require.Eventually(t, func() bool {
+		conns := getConnections(t, tr)
+		origConn = searchConnections(conns, func(cs network.ConnectionStats) bool {
```
Comment (leeavital) [inline pkg/network/tracer/tracer_linux_test.go:1714]: Does it make sense to have searchConnections use require.Eventually itself? this is the second such fix I've seen that we've fixed by wrapping searchConnections in a require.Eventually

File: pkg/network/tracer/tracer_linux_test.go [hunk 3]
```diff
@@ -1685,7 +1684,7 @@ func TestTCPDirectionWithPreexistingConnection(t *testing.T) {
 	// create an initial client connection to the server
 	c, err := net.DialTimeout("tcp", server.address, 5*time.Second)
 	require.NoError(t, err)
-	defer c.Close()
+	t.Cleanup(func() { c.Close() })
```
Comment (guyarb) [inline pkg/network/tracer/tracer_linux_test.go:1687]: nit: equivalent with "less code"
```suggestion
	t.Cleanup(c.Close)
```

Comment (hmahmood) [reply pkg/network/tracer/tracer_linux_test.go:1687]: `c.Close()` returns `error` so can't pass it to `t.Cleanup`

**Pull request merged**